### PR TITLE
Updating K8s docs

### PR DESCRIPTION
**Pull request by zmac12**
## Summary
By default, AWS deployments of K8s do not have access to S3 for persistent storage unless permissions are given. Updated the K8s docs to describe that behavior.

## Changes
Scaling out with Kubernetes docs. Addresses #3200 

## Importance
Users getting started with K8s on AWS might not understand the need for persistent storage and this update to the docs may simplify their getting started process.

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [x] adds a change file in the `changes/` directory (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

**Review by jcrist — COMMENTED** [review]
Thanks for the PR @zmac12. This feels like the wrong place to put this info (not all users use AWS, and the paragraph you edited was talking about docker image registry access, not S3). Looking through our docs, I think the proper place would be in the k8s agent docs. Could you add a new section on permissions before the `labels` section here: https://docs.prefect.io/orchestration/agents/kubernetes.html#labels?

**Comment by zmac12** [comment]
@jcrist Thanks for reviewing my PR! Let me know if this location is preferable or not and I can change it if needs be.

File: docs/orchestration/agents/kubernetes.md [hunk 1]
```diff
@@ -163,6 +163,10 @@ prefect-agent-845798bb59-s7wxg   1/1     Running   0          5s
 
 You are now ready to run some flows!
 
+#### Permissions
+
+In AWS deployments using Amazon Elastic Kubernetes Service, S3 is not accessible by default to Amazon EKS. To enable S3 access by your kubernetes cluster on EKS, add the necessary permissions (AmazonS3FullAccess or AmazonS3ReadOnlyAccess) directly to the NodeInstanceRole used by aws-auth-cm.yaml after launching worker nodes and before applying aws-auth-cm.yaml with kubectl.
```
Comment (jcrist) [inline docs/orchestration/agents/kubernetes.md:168]: Can you add something in here to qualify that S3 isn't required? Something like "If you need S3 access, note that..." or something else. Not all users will be affected by this.

**Review by jcrist — COMMENTED** [review]
One last tweak, then I think this is good to go. Thanks!

**Comment by zmac12** [comment]
@jcrist Thanks for reviewing that again! I just submitted the changes you requested, let me know if it looks good.

**Review by jcrist — APPROVED** [review]
Thanks!

**Pull request merged**